src/Workspaces/Remote/ServiceHub/Services/BrokeredServiceBase.cs

```diff
@@ -112,6 +130,23 @@ protected ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> implement
             return RunServiceImplAsync(implementation, cancellationToken);
         }
 
+        protected ValueTask RunServiceWithSolutionAsync(
```

Comment:
>RunServiceWithSolutionAsync [](http://example.com/codeflow?start=28&length=27)

I'd call this RunServiceAsync - we'd have two overloads.

Reply:
k